File: SearchA2DMatrix/solution.py

def solve(matrix,target):
    res=True
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    if len(matrix) == 0:
        return False
    i=0
    j=len(matrix[0]) - 1
    while i< len(matrix) and j >= 0:
        if matrix[i][j] > target:
            j-=1
        elif matrix[i][j] < target:
            i+=1
        else:
            return True

    return False
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    return res

File: SearchA2DMatrix/Solution.java
class Solution {
    // TODO: CRIO_TASK_MODULE_L1_PROBLEMS
    // complete the below function implementation
    public boolean searchMatrix(int[][] matrix, int target) {
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        if(matrix.length == 0) {
            return false;
        }
        int i = 0, j = matrix[0].length - 1;
        while(i < matrix.length && j >= 0) {
            if(matrix[i][j] > target) {
                j--;
            } else if(matrix[i][j] < target) {
                i++;
            } else {
                return true;
            }
        }
        return false;
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        // CRIO_UNCOMMENT_START_MODULE_L1_PROBLEMS
        // return false;
        // CRIO_UNCOMMENT_END_MODULE_L1_PROBLEMS
    }
}
